server/src/main/java/org/elasticsearch/index/shard/StoreRecovery.java

```diff
@@ -403,13 +404,15 @@ private void internalRecoverFromStore(IndexShard indexShard) throws IndexShardRe
                     writeEmptyRetentionLeasesFile(indexShard);
                 }
                 // since we recover from local, just fill the files and size
+                final RecoveryState.Index index = recoveryState.getIndex();
                 try {
-                    final RecoveryState.Index index = recoveryState.getIndex();
                     if (si != null) {
                         addRecoveredFileDetails(si, store, index);
                     }
                 } catch (IOException e) {
                     logger.debug("failed to list file details", e);
+                } finally {
```

Comment:
Not sure I understand why this needs to be in a finally?

Reply:
IIRC an `IOException` here can happen if force-allocating a stale primary, but is not fatal. I didn't investigate further since it doesn't really matter for our purposes: we reuse all the local files anyway so the reserved space is zero.

Reply:
Still not sure why the finally block is necessary, since the catch clause for IOException does not rethrow?

It seems illogical that in this block a finally block is necessary, whereas in the next block it is not?

I mean it will work fine either way... So feel free to disregard this if you prefer.

Reply:
Oh, I see now. Fixed in cf23d43f59fb1330fd0ab791e3f772654b6825f1.